packages/admin/src/middleware.page.ts

```diff
@@ -56,14 +95,7 @@ export async function middleware(req: NextRequest) {
     res.headers.set('Cache-Control', 'no-store');
     return res;
   }
-  let url = getLoginUrl();
-  console.log('Middleware redirect URL: ' + url);
-  if (submissionDownloadPattern.test({ pathname: req.nextUrl.pathname })) {
-    url = url + req.nextUrl.pathname;
-    console.log('Getting submission export download redirect URL: ' + url);
-  }
-  console.log('Final redirect URL from admin middleware: ' + url);
-  return NextResponse.redirect(url);
+  return redirectToLogin(req);
```

Comment:
Confused why if we've established we have a valid admin session, we redirect through the login flow. Will investigate to see if we can skip out that step

Reply:
Just got confused because the codes a mess - refactored code to some well defined functions and guard clauses